go/src/koding/kites/kloud/machine/fixture_test.go

```diff
@@ -0,0 +1,179 @@
+package machine
+
+import (
+	"time"
+
+	"koding/db/models"
+	"koding/db/mongodb/modelhelper"
+
+	mgo "gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+)
+
+// fixture is a mock for MongoDB tables.
+type fixture struct {
+	MachineColl       []*models.Machine       // machines collection.
+	StackTemplateColl []*models.StackTemplate // stack template collection.
+}
+
+var _ adapter = (*fixture)(nil)
+
+// GetAllMachinesByUsername mocks GetMachinesByUsername method.
+func (f *fixture) GetAllMachinesByUsername(username string) (ms []*models.Machine, err error) {
+	for _, m := range f.MachineColl {
+		for _, u := range m.Users {
+			if u.Username == username {
+				ms = append(ms, m)
+				break
+			}
+		}
+	}
+
+	if len(ms) == 0 {
+		return nil, mgo.ErrNotFound
+	}
+
+	return ms, nil
+}
+
+// GetStackTemplateFieldsByIds mocks GetStackTemplateFieldsByIds method.
+func (f *fixture) GetStackTemplateFieldsByIds(ids []bson.ObjectId, _ []string) (sts []*models.StackTemplate, err error) {
+	for _, st := range f.StackTemplateColl {
+		for _, id := range ids {
+			if id == st.Id {
+				sts = append(sts, st)
+				break
+			}
+		}
+	}
+
+	if len(sts) == 0 {
+		return nil, mgo.ErrNotFound
+	}
+
+	return sts, nil
+}
+
+// Fix1 describes the following database state:
+//
+// Team: orange,
+// Users:
+//   - bober:
+//      - bober-aws-0, stack: boberStack, admin.
+//      - bober-aws-1, stack: boberStack, admin, shared with: john, blaster.
+//   - john:
+//      - john-google-0, stack: johnStack, shared with: bober, blaster(not approved).
+//   - blaster:
+//      - blaster-aws-0, stack: blasterStack.
+//      - instance from koding provider.
+//
+var Fix1 = &fixture{
```

Comment:
Mostly, i am against fixtures, since you dont test the actual code communicating with the DB, we dont have any kind of fixture data in memory in socialapi and we had great success with it. Whenever you change any property in the schema, you will have to maintain these fixtures as well. Yes you can tell also many good things about having fixtures ie: faster, but working with the whole system  is always better. I would suggest you to use the database and remove all the helper functionality for fixtures.